### ISSUE TITLE
j2v8 dependency is not really optional

### ISSUE THREAD

**Issue by Clashsoft** [nidi3/graphviz-java#146]
When using the `graphviz-java-no-j2v8` artifact, **or even the `graphviz-java` one without specifying an additional dependency on j2v8**, I get the following error:

```
java.lang.NoClassDefFoundError: com/eclipsesource/v8/V8RuntimeException
        at guru.nidi.graphviz.engine.GraphvizV8Engine.lambda$new$41(GraphvizV8Engine.java:26)
        at guru.nidi.graphviz.engine.AbstractJsGraphvizEngine.engine(AbstractJsGraphvizEngine.java:51)
        at guru.nidi.graphviz.engine.AbstractJsGraphvizEngine.doInit(AbstractJsGraphvizEngine.java:42)
        at guru.nidi.graphviz.engine.AbstractGraphvizEngine.initTask(AbstractGraphvizEngine.java:53)
        at guru.nidi.graphviz.engine.AbstractGraphvizEngine.init(AbstractGraphvizEngine.java:45)
        at guru.nidi.graphviz.engine.Graphviz.doUseEngine(Graphviz.java:100)
        at guru.nidi.graphviz.engine.Graphviz.lambda$doUseEngine$9(Graphviz.java:100)
        at guru.nidi.graphviz.engine.AbstractGraphvizEngine.initTask(AbstractGraphvizEngine.java:58)
        at guru.nidi.graphviz.engine.AbstractGraphvizEngine.init(AbstractGraphvizEngine.java:45)
        at guru.nidi.graphviz.engine.Graphviz.doUseEngine(Graphviz.java:100)
        at guru.nidi.graphviz.engine.Graphviz.useEngine(Graphviz.java:92)
        at guru.nidi.graphviz.engine.Graphviz.useEngine(Graphviz.java:73)
        at guru.nidi.graphviz.engine.Graphviz.useDefaultEngines(Graphviz.java:65)
        at guru.nidi.graphviz.engine.Graphviz.getEngine(Graphviz.java:106)
        at guru.nidi.graphviz.engine.Graphviz.execute(Graphviz.java:223)
        at guru.nidi.graphviz.engine.Renderer.execute(Renderer.java:133)
        at guru.nidi.graphviz.engine.Renderer.toFile(Renderer.java:54)
        at org.fulib.tools.ObjectDiagrams.dump(ObjectDiagrams.java:202)
        at org.fulib.tools.ObjectDiagrams.dumpPng(ObjectDiagrams.java:86)
        at org.fulib.patterns.TestStudyRightPatterns.testPatterns(TestStudyRightPatterns.java:60)
        Caused by:
        java.lang.ClassNotFoundException: com.eclipsesource.v8.V8RuntimeException
            at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
            at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
            at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
```

This suggests that the dependency may not be "optional" at all. I believe either it needs to be made required again, or exception handling in `GraphvizV8Engine` should be adapted so that it catched the NoClassDefFound/ClassNotFound exceptions and uses a different engine instead. Alternatively, you should specify in the README that when the j2v8 dependency is not manually added, one must use `Graphviz.useEngine(...)`.

Important to note: I am using Gradle. Maybe it resolves dependencies differently.

---

# Workaround:

The "lazy" solution for Gradle users is to declare a dependency on `graphviz-java-all-j2v8`. This makes the already quite large dependency even larger, though. There may also be a way to select the right j2v8 artifact based on the OS in `build.gradle`, but that doesn't work well for libraries.

**Comment by nidi3** [comment]
`graphviz-java-no-j2v8` looks definitvely like a problem.
`graphviz-java` probably is a gradle issue. With maven, the version corresponding to your platform (linux, max, win) is depended on with a profile. Gradle seems to ignore this profile.

**Comment by Clashsoft** [comment]
> Gradle seems to ignore this profile.

I found this in the [Gradle docs](https://docs.gradle.org/current/userguide/migrating_from_maven.html#migmvn:profiles_and_properties):

> And since Gradle has much more powerful ways to declare conditions, it does not need to have formal support for profiles (**except in the POMs of dependencies**).

So it should support this. Didn't find any specifics about the highlighted part, though.

**Comment by nidi3** [comment]
I looked at how JavaFx is doing it, they need a plugin:
https://openjfx.io/openjfx-docs/#gradle
So for the moment, I don't see a good solution, but just documenting this in readme.

**Comment by Clashsoft** [comment]
Oh well, but I guess the note in the readme plus the extra runtime checks are enough. Thanks a lot!

**Comment by nidi3** [comment]
fixed in version `0.15.1`
